functions/src/index.ts

```diff
@@ -0,0 +1,172 @@
+    }
+    else {
+        team2UpdateMap.set('wins', increment_value);
+        team2UpdateMap.set('stats.' + resTeam2 + ':' + resTeam1, increment_value);
+        team1UpdateMap.set('losses', increment_value);
+        team1UpdateMap.set('stats.' + resTeam1 + ':' + resTeam2, increment_value);
+    }
+
+    let preTeams: string[] = [];
+    snap.data().dominations.forEach((dTeam: DocumentReference<DocumentData>) => {
+        if (!preTeams.includes(dTeam.id)) { 
+            if (dTeam.id === team1Id) {
+                team1UpdateMap.set('dominations', increment_value);
+            }
+            else {
+                team2UpdateMap.set('dominations', increment_value);
+            }
+            preTeams.push(dTeam.id);
+        }
+    });
+
+    preTeams = [];
+    snap.data().defeats.forEach((dTeam: DocumentReference<DocumentData>) => {
+        if (!preTeams.includes(dTeam.id)) { 
+            if (dTeam.id === team1Id) {
+                team1UpdateMap.set('defeats', increment_value);
+            }
+            else {
+                team2UpdateMap.set('defeats', increment_value);
+            }
+            preTeams.push(dTeam.id);
+        }
+    });
+
+    const team1UpdateObject = Object.fromEntries(team1UpdateMap);
+    const team2UpdateObject = Object.fromEntries(team2UpdateMap);
+
+    db.doc(prefix + 'Teams/' + team1Id).update(team1UpdateObject);
+    db.doc(prefix + 'Teams/' + team2Id).update(team2UpdateObject);
+    db.doc(prefix + 'Users/' + playersTeam1[0].id).update(team1UpdateObject);
+    db.doc(prefix + 'Users/' + playersTeam1[1].id).update(team1UpdateObject);
+    db.doc(prefix + 'Users/' + playersTeam2[0].id).update(team2UpdateObject);
+    db.doc(prefix + 'Users/' + playersTeam2[1].id).update(team2UpdateObject);
+}
+
+/* Update stats for Users and Teams when match is added to firestore in testing environment */
+export const updateUsersAndTeamsT = functions.firestore.document(`T-Matches/{documentId}`).onCreate((snap, context) => {
+    updateUsersAndTeamsProcedure(snap);
+    return null;
+});
+
+/* Update stats for Users and Teams when match is added to firestore in production environment */
+export const updateUsersAndTeams = functions.firestore.document(`Matches/{documentId}`).onCreate((snap, context) => {
+    updateUsersAndTeamsProcedure(snap);
+    return null;
+});
+
+/* When a new user is added to firebase, this function gets triggered.
+   It adds a new user to firestore. */
+export const newUser = functions.auth.user().onCreate((user) => {
+    const prefix = process.env.PREFIX;
+
+    const newUser: IUser = {
+        id: user.uid,
+        name: user.displayName!,
+        wins: 0,
+        losses: 0,
+        defeats: 0,
+        dominations: 0,
+        stats: {
+          '0:2': 0,
+          '2:0': 0,
+          '1:2': 0,
+          '2:1': 0,
+        },
+    };
+    db.doc(prefix + 'Users/' + user.uid).set(newUser);
+
+    return null;
+});
```

Comment:
Can we maybe split the functions into more than one file? Because it makes more sense to have a file for the new user process and another file for the addMatch process?